go-controller/pkg/ovn/egressfirewall_test.go

```diff
@@ -1014,13 +1108,122 @@ var _ = ginkgo.Describe("OVN EgressFirewall Operations", func() {
 						return retry.CheckRetryObj(efKey, fakeOVN.controller.retryEgressFirewalls)
 					}, time.Second).Should(gomega.BeFalse())
 
+					if !oldDNS {
+						// delete the dns name resolver object.
+						err = fakeOVN.fakeClient.OCPNetworkClient.NetworkV1alpha1().DNSNameResolvers(config.Kubernetes.OVNConfigNamespace).
+							Delete(context.TODO(), "dns-default", metav1.DeleteOptions{})
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					}
+
 					// check dns address set is cleaned up on delete
 					gomega.Eventually(fakeOVN.nbClient).Should(libovsdbtest.HaveData(initialData))
 					return nil
 				}
 				err := app.Run([]string{app.Name})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-			})
+			},
+				table.Entry(fmt.Sprintf("correctly cleans up object that failed to be created using old dns resolution, gateway mode %s", gwMode), gwMode, true),
+				table.Entry(fmt.Sprintf("correctly cleans up object that failed to be created using new dns resolution, gateway mode %s", gwMode), gwMode, false),
+			)
+			table.DescribeTable("correctly creates egress firewall using different dns resolution methods and ip families", func(gwMode config.GatewayMode, oldDNS bool, resolvedIP string) {
+				if !oldDNS {
+					// enable the dns name resolver flag.
+					config.OVNKubernetesFeature.EnableDNSNameResolver = true
+				}
+				config.Gateway.Mode = gwMode
+				app.Action = func(ctx *cli.Context) error {
+					namespace1 := *newNamespace("namespace1")
+					dnsName := "a.b.c"
```

Comment:
should we have one more test for the new resolver only with a wildcard dns name?